rotkehlchen/exchanges/bitfinex.py

```diff
@@ -654,23 +641,29 @@ def _query_currency_map(self) -> CurrencyMapResponse:
                     f'{self.name} currency map returned invalid JSON response. Check further logs',
                 )
             else:
-                currency_map = {
-                    bfx_symbol: symbol for bfx_symbol, symbol in response_list[0]
-                    if bfx_symbol not in set(BITFINEX_EXCHANGE_TEST_ASSETS)
-                }
-                with GlobalDBHandler().conn.read_ctx() as cursor:
-                    cursor.execute(
-                        'SELECT exchange_symbol, local_id FROM location_asset_mappings WHERE location IS ? OR location IS NULL;',  # noqa: E501
-                        (Location.BITFINEX.serialize_for_db(),),
-                    )
-                    for bfx_symbol, asset_id in cursor:
-                        currency_map[bfx_symbol] = asset_id
+                test_assets = set(BITFINEX_EXCHANGE_TEST_ASSETS)
+                bitfinex_db_serialized = Location.BITFINEX.serialize_for_db()
+                for bfx_symbol, symbol in response_list[0]:
+                    with GlobalDBHandler().conn.read_ctx() as cursor:
+                        if cursor.execute(
+                            'SELECT COUNT(*) FROM location_unsupported_assets WHERE location=? AND exchange_symbol=?',  # noqa: E501
+                            (bitfinex_db_serialized, bfx_symbol),
+                        ).fetchone()[0] > 0 or bfx_symbol in test_assets:
+                            continue
+
+                    try:
+                        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                            write_cursor.execute(
+                                'INSERT OR IGNORE INTO location_asset_mappings (location, exchange_symbol, local_id) VALUES (?, ?, ?);', (  # noqa: E501
+                                    bitfinex_db_serialized,
+                                    bfx_symbol,
+                                    symbol_to_asset_or_token(symbol).serialize(),
```

Comment:
It feels like doing `symbol_to_asset_or_token` it is skipping the known mappings for bitfinex and you are doing extra job that you shouldn't be doing? Am I missing something?